src/storage/v2/inmemory/edge_type_index.cpp

```diff
@@ -208,6 +208,12 @@ void InMemoryEdgeTypeIndex::Iterable::Iterator::AdvanceUntilValid() {
     auto *from_vertex = index_iterator_->from_vertex;
     auto *to_vertex = index_iterator_->to_vertex;
 
+    if (self_->transaction_->original_start_timestamp.has_value()) {
+      if (index_iterator_->timestamp >= self_->transaction_->original_start_timestamp.value()) {
+        continue;
+      }
+    }
+
```

Comment:
Can we merge this with the `IsEdgeVisible`?

Reply:
It's a bit tricky since this is not tied to the mvcc logic, but to keeping the initial iterator from indices in place. I have made the logic inside one place in `indices_utils.hpp`.
The `IsEdgeVisible` is again called during edge creation so I'm not even sure what to pass to the edge at that point, which is a correct timestamp or not. And even if I introduce an optional, it seems like there is a lot going on in that method.